### PR TITLE
Implement the 'visibility' API for Android projects

### PR DESCRIPTION
**Pull request by ifarbod**
As discussed in https://github.com/premake/premake-core/issues/1251#issuecomment-479895565

### CODE REVIEW

File: modules/android/vsandroid_vcxproj.lua [hunk 1]
```diff
@@ -338,6 +338,17 @@
...
 
+			-- -fvisibility=<>
+			if cfg.visibility ~= nil then
+				local visibility_map = {
+					["Default"] 	= "default",
+					["Hidden"] 		= "hidden",
+					["Internal"] 	= "internal",
+					["Protected"] 	= "protected",
+				}
+				table.insert(opts, '-fvisibility=' .. visibility_map[cfg.visibility])
```
Comment (ratzlaff) [inline modules/android/vsandroid_vcxproj.lua]: This mapping exists in the [gcc tool](https://github.com/premake/premake-core/blob/master/src/tools/gcc.lua#L193). Does it need to be duplicated here?

Comment (ifarbod) [reply modules/android/vsandroid_vcxproj.lua]: Can I access it like this? `gcc.cxxflags.visibility` ?

Comment (ifarbod) [reply modules/android/vsandroid_vcxproj.lua]: I just tried it and I'm getting an error: `Error: [string "android/vsandroid_vcxproj.lua"]:268: attempt to index a nil value (global 'gcc')`

Comment (ifarbod) [reply modules/android/vsandroid_vcxproj.lua]: I think it's better now https://github.com/ifarbod/premake-core/commit/044cddd81a00e65a82e4eada63100b3ff799e60c

**Comment by ifarbod** [comment]
@samsinsane There's the "Squash and merge" option on the dropdown next to the Merge button but sure, I'll squash them anyway.

**Pull request merged**